messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -145,15 +146,26 @@ object GistSdk : Application.ActivityLifecycleCallbacks {
     fun showMessage(message: Message, position: MessagePosition? = null): String? {
         ensureInitialized()
         var messageShown = false
-
-        GlobalScope.launch {
-            try {
-                messageShown = gistModalManager.showModalMessage(message, position)
-            } catch (e: Exception) {
-                Log.e(GIST_TAG, "Failed to show message: ${e.message}", e)
+        val queueId = message.queueId
+
+        if (shownMessageQueueIds.contains(queueId)) {
+            Log.i(GIST_TAG, "Duplicate message $queueId skipped")
+        } else {
+            GlobalScope.launch {
+                try {
+                    messageShown = gistModalManager.showModalMessage(message, position)
+                } catch (e: Exception) {
+                    Log.e(GIST_TAG, "Failed to show message: ${e.message}", e)
+                }
             }
         }
-        return if (messageShown) message.instanceId else null
+
+        if (messageShown) {
+            shownMessageQueueIds.add(queueId!!)
```

Comment:
Have you tested launching a message programmatically? Wouldn't this crash the app if queueId is not present?